shared/src/commonMain/kotlin/com/intive/picover/auth/repository/AuthRepository.kt

```diff
@@ -32,47 +31,37 @@ class AuthRepository(
 	}
 
 	suspend fun userProfile() =
-		try {
-			val photoUrl = userAvatarReference.getDownloadUrl()
-			Result.success(createProfile(photoUrl))
-		} catch (storageException: FirebaseStorageException) {
-			Result.success(createProfile(null))
-		} catch (exception: Exception) {
-			Result.failure(exception)
+		runCatching {
+			currentUserProfile()
 		}
 
 	suspend fun deleteAccount() =
 		try {
-			requireUser().delete()
+			currentUser.delete()
 			AccountDeletionResult.Success
 		} catch (firebaseAuthRecentLoginRequiredException: FirebaseAuthRecentLoginRequiredException) {
 			firebaseAuth.signOut()
 			AccountDeletionResult.ReAuthenticationNeeded
 		}
 
 	suspend fun updateUserAvatar(file: File) =
-		try {
-			userAvatarReference.putFile(file)
-			userProfile()
-		} catch (exception: Exception) {
-			Result.failure(exception)
+		runCatching {
+			val photoReference = storageReference.child("user/${currentUser.uid}")
+			photoReference.putFile(file)
+			currentUser.updateProfile(photoUrl = photoReference.getDownloadUrl())
+			currentUserProfile()
 		}
 
 	suspend fun updateUserName(userName: String) =
-		try {
-			requireUser().updateProfile(displayName = userName)
-			userProfile()
-		} catch (exception: Exception) {
-			Result.failure(exception)
+		runCatching {
+			currentUser.updateProfile(displayName = userName)
+			currentUserProfile()
 		}
 
-	private fun requireUser() =
-		firebaseAuth.currentUser!!
-
-	private fun createProfile(photoUrl: String?) =
-		requireUser().let {
+	private fun currentUserProfile() =
+		currentUser.let {
 			Profile(
-				photo = photoUrl,
+				photo = it.photoURL,
```

Comment:
Photo was referenced from firebase storage each time. Now, the URL is stored in firebase user